can't compile for ARM (including AArch64)

(will be closed by #530)

```
error[E0308]: mismatched types
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/llama-cpp-2-0.1.81/src/model.rs:560:32
    |
560 |                 CStr::from_ptr(buff.as_mut_ptr())
    |                 -------------- ^^^^^^^^^^^^^^^^^ expected `*const u8`, found `*mut i8`
    |                 |
    |                 arguments to this function are incorrect
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*mut i8`
note: associated function defined here
   --> /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ffi/c_str.rs:282:25

For more information about this error, try `rustc --explain E0308`.
error: could not compile `llama-cpp-2` (lib) due to 1 previous error
```

The existing code in `src/model.rs` (`LlamaModel::apply_chat_template`) uses `i8` for all platforms / targets, but some targets (e.g. `aarch64-unknown-linux-gnu` was the target for the logs above) have `u8` as the C `char` type: https://doc.rust-lang.org/1.81.0/src/core/ffi/mod.rs.html#95-157

--- a/llama-cpp-2/src/model.rs
+++ b/llama-cpp-2/src/model.rs
@@ -523,7 +523,7 @@ impl LlamaModel {
         let message_length = chat.iter().fold(0, |acc, c| {
             acc + c.role.to_bytes().len() + c.content.to_bytes().len()
         });
-        let mut buff: Vec<i8> = vec![0_i8; message_length * 4];
+        let mut buff = vec![0; message_length * 4];
 
         // Build our llama_cpp_sys_2 chat messages
         let chat: Vec<llama_cpp_sys_2::llama_chat_message> = chat
@@ -548,7 +548,7 @@ impl LlamaModel {
                 chat.as_ptr(),
                 chat.len(),
                 add_ass,
-                buff.as_mut_ptr().cast::<std::os::raw::c_char>(),
+                buff.as_mut_ptr(),
                 buff.len() as i32,
             );
             // A buffer twice the size should be sufficient for all models, if this is not the case for a new model, we can increase it

